G1ANT.Addon.Net/Models/AttachmentModel.cs

```diff
@@ -0,0 +1,44 @@
+﻿using MailKit;
+using MimeKit;
+
+namespace G1ANT.Addon.Net.Models
+{
+    public class AttachmentModel : IAttachmentModel
+    {
+        public string Name { get; }
+        public long Size { get; }
+        public string Type { get; }
+        public MimeEntity MimeEntity { get; }
+
+        const string emptyAttachmentDefaultName = "unnamed-attachment";
+
+        public AttachmentModel(MimeEntity mimeEntity)
+        {
+            Name = GetAttachmentName(mimeEntity);
+            Size = GetAttachmentSize(mimeEntity);
+            Type = GetAttachmentMediaTypeName(mimeEntity);
+            MimeEntity = mimeEntity;
```

Comment:
Setting object state should be its first activity, otherwise some methods called in constructor before object will be initialized might fail. This is also a pattern that might lead to bug introduction: you might expect that all fields/properties of the model are properly initialized during execution of your method, while in fact they are not.